### PR TITLE
ROC 2835 restrict file types

### PR DESCRIPTION
**Pull request by robertparkinson**

### CODE REVIEW

File: src/main/features/certificateOfService/routes/file-upload.ts [hunk 1]
```diff
@@ -17,29 +19,47 @@ export default express.Router()
...
-          fs.unlink(file.path, function (err) {
-            if (err) {
-              next(err)
-            }
+        if (file.size === 0) {
+          user.legalUploadDocumentDraft.document.fileToUploadError = FileTypeErrors.FILE_REQUIRED
+          new DraftService().save(user.legalUploadDocumentDraft, user.bearerToken).then(() => {
+            res.redirect(Paths.documentUploadPage.uri)
+          })
+        } else if (file.size > 10000000) {
```
Comment (sabahirfan) [inline src/main/features/certificateOfService/routes/file-upload.ts]: can we please extract this as a const?

File: src/main/features/certificateOfService/routes/file-upload.ts [hunk 2]
```diff
@@ -17,29 +19,47 @@ export default express.Router()
...
-          user.legalCertificateOfServiceDraft.document.uploadedDocuments.map((file) => files.push(new UploadedDocument().deserialize(file)))
-          files.push(new UploadedDocument(file.name, file.type, documentType, documentManagementURI))
-
-          fs.unlink(file.path, function (err) {
-            if (err) {
-              next(err)
-            }
+        if (file.size === 0) {
+          user.legalUploadDocumentDraft.document.fileToUploadError = FileTypeErrors.FILE_REQUIRED
+          new DraftService().save(user.legalUploadDocumentDraft, user.bearerToken).then(() => {
```
Comment (sabahirfan) [inline src/main/features/certificateOfService/routes/file-upload.ts]: This seems duplicated in all blocks, can we please extract this out

File: src/main/features/certificateOfService/routes/what-documents.ts [hunk 3]
```diff
@@ -30,6 +30,14 @@ export default express.Router()
         user.legalUploadDocumentDraft.document.fileToUpload = undefined
         await new DraftService().save(res.locals.user.legalUploadDocumentDraft, res.locals.user.bearerToken)
 
+        if (res.locals.user.legalUploadDocumentDraft && res.locals.user.legalUploadDocumentDraft['id']) {
+          await new DraftService()['delete'](res.locals.user.legalUploadDocumentDraft['id'], res.locals.user.bearerToken)
```
Comment (sabahirfan) [inline src/main/features/certificateOfService/routes/what-documents.ts]: is it possible to call like `new DraftService().delete(...)` ?

File: src/main/features/certificateOfService/routes/file-upload.ts (hunk 1 above)
Comment (robertparkinson) [reply src/main/features/certificateOfService/routes/file-upload.ts]: done

File: src/main/features/certificateOfService/routes/file-upload.ts (hunk 2 above)
Comment (robertparkinson) [reply src/main/features/certificateOfService/routes/file-upload.ts]: done

File: src/main/features/certificateOfService/routes/what-documents.ts (hunk 3 above)
Comment (robertparkinson) [reply src/main/features/certificateOfService/routes/what-documents.ts]: they shouldn't have been there. was in there for testing. Have removed

File: src/main/features/certificateOfService/routes/file-upload.ts [hunk 4]
```diff
@@ -8,36 +8,48 @@ import { DocumentType } from 'forms/models/documentType'
...
 import User from 'idam/user'
 import ErrorHandling from 'common/errorHandling'
+import { FileTypes } from 'forms/models/fileTypes'
+import { FileTypeErrors } from 'forms/models/fileTypeErrors'
 
 export default express.Router()
   .post(Paths.fileUploadPage.uri,
     ErrorHandling.apply(async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
       const form = new formidable.IncomingForm()
+      const FILE_SIZE_LIMIT: number = 10000000
```
Comment (sabahirfan) [inline src/main/features/certificateOfService/routes/file-upload.ts]: Shouldn't this be 10485760?

**Pull request merged**